subjects/PE/x86/VCExeSample/VCExeSample.c

```diff
@@ -92,36 +91,24 @@ void const_div_test10(word32 dwArg04)
 	globals->dw403020 = eax_16;
 }
 
-// 004012D0: void loop_test11(Register word32 ecx, Register word32 ebp)
-void loop_test11(word32 ecx, word32 ebp)
+// 004012D0: void loop_test11(Stack real64 rArg04)
+void loop_test11(real64 rArg04)
 {
-	struct Eq_218 * ebp_19 = fp - 0x04;
-	while (*(ebp_19 - 0x04) > 0x00)
+	int32 dwLoc08_28 = 0x05;
+	while (dwLoc08_28 > 0x00)
 	{
-		ui32 eax_26 = *(ebp_19 - 0x04);
-		ui32 eax_27 = eax_26 & 0x80000001;
-		if ((eax_26 & 0x80000001) < 0x00)
-			eax_27 = ((eax_26 & 0x80000001) - 0x01 | ~0x01) + 0x01;
-		if (eax_27 == 0x00)
-		{
-			*(ebp_19 - 0x08) = (real32) ebp_19->r0008;
-			real64 rLoc1_44 = (real64) *(ebp_19 - 0x08);
-			*(fp - 0x10) = ecx;
-			*(fp - 0x10) = (real32) rLoc1_44;
-			loop_test9(rArg00, rArg0);
-		}
+		int32 eax_32 = dwLoc08_28 & 0x80000001;
+		if (eax_32 < 0x00)
+			eax_32 = (eax_32 - 0x01 | ~0x01) + 0x01;
+		if (eax_32 == 0x00)
+			loop_test9((real32) rArg04, rArg0);
 		else
-		{
-			*(fp - 0x14) = ebp_19->r0008;
-			ebp_19 = nested_if_blocks_test8(rArg00, rArg0);
-		}
-		word32 ecx_38 = *(ebp_19 - 0x04);
-		*(ebp_19 - 0x04) = ecx_38 - 0x01;
-		ecx = ecx_38 - 0x01;
+			nested_if_blocks_test8(rArg04, rArg0);
+		dwLoc08_28 = dwLoc08_28 - 0x01;
```

Comment:
This is big improvement. Is not it, John?

Reply:
Absolutely. We don't want to lose calls after all!